src/vs/base/parts/quickinput/browser/quickInput.ts

```diff
@@ -487,9 +487,19 @@ class QuickPick<T extends IQuickPickItem> extends QuickInput implements IQuickPi
 	}
 
 	set value(value: string) {
+		this.doSetValue(value);
+	}
+
+	private doSetValue(value: string, skipUpdate?: boolean): void {
 		if (this._value !== value) {
-			this._value = value || '';
-			this.update();
+			this._value = value;
+			if (!skipUpdate) {
+				this.update();
+			}
+			const didFilter = this.ui.list.filter(this.filterValue(this._value));
```

Comment:
The UI should only be updated if this QuickPick is visible (another QuickPick could be visible).

Reply:
@chrmarti 👍 , see https://github.com/microsoft/vscode/pull/137857/commits/f00d30357fb835e272e6e244b836a9e235482486